### PR TITLE
Extra logging and hmmsearch fix

### PR DESCRIPTION
**Pull request by malanjary-wur**
Latest updates to improve stability

### CODE REVIEW

File: cblaster/hmm_search.py [hunk 1]
```diff
@@ -131,7 +134,13 @@ def write_profiles(profiles: Collection[str], output: str=None) -> str:
         output: name of output file
     """
     if not output:
-        output = datetime.now().strftime("cblaster_%Y%m%d%H%M%S.hmm")
+        counter = 0
+        output = Path(cagecat_prefix, datetime.now().strftime(f"cblaster_%Y%m%d%H%M%S-{counter}.hmm"))
```
Comment (gamcil) [inline cblaster/hmm_search.py:138]: Can we generalise this and take out the hardcoded CAGECAT stuff here? (e.g. add an output folder to the arguments for `write_profiles`)

File: cblaster/hmm_search.py [hunk 2]
```diff
@@ -149,7 +158,10 @@ def run_hmmsearch(fasta, query):
         temp_res: List, String of result file names
     """
     LOG.info("Performing hmmsearch")
-    output = Path(query).with_suffix(".txt")
+    output = Path(cagecat_prefix, query).with_suffix(".txt")
```
Comment (gamcil) [inline cblaster/hmm_search.py:161]: Here as well, on the CAGECAT side can you just have `cagecat_prefix` as part of `query` when you pass it to the function?

**Review by gamcil — CHANGES_REQUESTED** [review]
Sorry it took a while to get around to looking at this PR. Mostly happy with everything, would just like to take out the specific CAGECAT stuff.

**Comment by malanjary-wur** [comment]
Hi Cam,

I'll take a look and see if try and get you a new pull request asap.

Best,
-Mohammad

On Feb 14, 2023, at 4:20 AM, Cameron Gilchrist ***@***.******@***.***>> wrote:



@gamcil requested changes on this pull request.

Sorry it took a while to get around to looking at this PR. Mostly happy with everything, would just like to take out the specific CAGECAT stuff.

________________________________

In cblaster/hmm_search.py<https://eur03.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fgamcil%2Fcblaster%2Fpull%2F87%23discussion_r1105248493&data=05%7C01%7Cmohammad.alanjary%40wur.nl%7C7288535ba1a4484201ef08db0e3a7a07%7C27d137e5761f4dc1af88d26430abb18f%7C0%7C0%7C638119416549291395%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=5lc3tiDB2Z%2F032gLO2cLQKEpm9or4veaObPtwCheTY4%3D&reserved=0>:

> @@ -131,7 +134,13 @@ def write_profiles(profiles: Collection[str], output: str=None) -> str:
         output: name of output file
     """
     if not output:
-        output = datetime.now().strftime("cblaster_%Y%m%d%H%M%S.hmm")
+        counter = 0
+        output = Path(cagecat_prefix, datetime.now().strftime(f"cblaster_%Y%m%d%H%M%S-{counter}.hmm"))


Can we generalise this and take out the hardcoded CAGECAT stuff here? (e.g. add an output folder to the arguments for write_profiles)

________________________________

In cblaster/hmm_search.py<https://eur03.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fgamcil%2Fcblaster%2Fpull%2F87%23discussion_r1105249040&data=05%7C01%7Cmohammad.alanjary%40wur.nl%7C7288535ba1a4484201ef08db0e3a7a07%7C27d137e5761f4dc1af88d26430abb18f%7C0%7C0%7C638119416549291395%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=P2H%2BmEWjSUoOsQ%2FQs%2B6Zpu%2BWYHCH2Xp0AvzfBRSTWCI%3D&reserved=0>:

> @@ -149,7 +158,10 @@ def run_hmmsearch(fasta, query):
         temp_res: List, String of result file names
     """
     LOG.info<http://LOG.info>("Performing hmmsearch")
-    output = Path(query).with_suffix(".txt")
+    output = Path(cagecat_prefix, query).with_suffix(".txt")


Here as well, on the CAGECAT side can you just have cagecat_prefix as part of query when you pass it to the function?

**Comment by malanjary-wur** [comment]
Hi Cam,

I sent Matthias a notice about editing the cagecat specific stuff out as I think he might be able to resolve this quicker. In other related news I have unfortunately seen about 20 failed jobs on the CAGECAT site and it seems there is a small bug that I have a temporary fix for.

The problem:
When writing the _summary.txt file for some reason there is a ‘None’ value for the cluster accession in a row. This breaks the formatter in line 17 formatters.py (can’t format the None value). See attached session file if you want to reproduce (when writing session.format(’summary’ ….) )


Solution:
I have added a quick fix which seems to remedy this but may have some other consequences you might be more aware of. Right now it will show “None” in subject column for an item that has this issue.

Just ensured it is parsed as a str( )...

fmt = [f"{str(row[index]):{length}}" for index, length in enumerate(lengths)]

So far I am not getting jobs with this error.

Considering its a small fix I thought maybe not worth a whole other pull request.

Regards,
-Mohammad

On Feb 14, 2023, at 4:20 AM, Cameron Gilchrist ***@***.******@***.***>> wrote:



@gamcil requested changes on this pull request.

Sorry it took a while to get around to looking at this PR. Mostly happy with everything, would just like to take out the specific CAGECAT stuff.

________________________________

In cblaster/hmm_search.py<https://eur03.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fgamcil%2Fcblaster%2Fpull%2F87%23discussion_r1105248493&data=05%7C01%7Cmohammad.alanjary%40wur.nl%7C7288535ba1a4484201ef08db0e3a7a07%7C27d137e5761f4dc1af88d26430abb18f%7C0%7C0%7C638119416549291395%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=5lc3tiDB2Z%2F032gLO2cLQKEpm9or4veaObPtwCheTY4%3D&reserved=0>:

> @@ -131,7 +134,13 @@ def write_profiles(profiles: Collection[str], output: str=None) -> str:
         output: name of output file
     """
     if not output:
-        output = datetime.now().strftime("cblaster_%Y%m%d%H%M%S.hmm")
+        counter = 0
+        output = Path(cagecat_prefix, datetime.now().strftime(f"cblaster_%Y%m%d%H%M%S-{counter}.hmm"))


Can we generalise this and take out the hardcoded CAGECAT stuff here? (e.g. add an output folder to the arguments for write_profiles)

________________________________

In cblaster/hmm_search.py<https://eur03.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fgamcil%2Fcblaster%2Fpull%2F87%23discussion_r1105249040&data=05%7C01%7Cmohammad.alanjary%40wur.nl%7C7288535ba1a4484201ef08db0e3a7a07%7C27d137e5761f4dc1af88d26430abb18f%7C0%7C0%7C638119416549291395%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=P2H%2BmEWjSUoOsQ%2FQs%2B6Zpu%2BWYHCH2Xp0AvzfBRSTWCI%3D&reserved=0>:

> @@ -149,7 +158,10 @@ def run_hmmsearch(fasta, query):
         temp_res: List, String of result file names
     """
     LOG.info<http://LOG.info>("Performing hmmsearch")
-    output = Path(query).with_suffix(".txt")
+    output = Path(cagecat_prefix, query).with_suffix(".txt")


Here as well, on the CAGECAT side can you just have cagecat_prefix as part of query when you pass it to the function?

**Comment by gamcil** [comment]
Yeah I have noticed that error and have made pretty much the exact same fix for the next release. 

It might just be quicker if I edit the PR directly - I think the easiest way will be to just use temporary files here, which will also remove the need to generate the unique name from a timestamp. Do you do anything special on the CAGECAT side with the HMM files or just delete them post search?

**Comment by malanjary-wur** [comment]
Hi Cam,

Great that solution sounds good to me, we just remove them after sometime so having them in temp works for CAGECAT. Maybe it can be something configureable if it's easy? (Some users might have limited space not he /tmp drive… but the files are not very large so probably not something to worry about. I liked using the “tempfile” package worked well for other projects on all operating systems, automatically seeks out an OS’s tmp location too.

Best,
-Mohammad

On Feb 16, 2023, at 2:51 AM, Cameron Gilchrist ***@***.******@***.***>> wrote:



Yeah I have noticed that error and have made pretty much the exact same fix for the next release.

It might just be quicker if I edit the PR directly - I think the easiest way will be to just use temporary files here, which will also remove the need to generate the unique name from a timestamp. Do you do anything special on the CAGECAT side with the HMM files or just delete them post search?

**Pull request merged**